knowledge-base/grid-binding-to-expando-object.md

```diff
@@ -188,82 +185,395 @@ In addition to the sample below, there is a [complete runnable project in GitHub
+The following list contains information about requirements and limitations when binding the Grid to `ExpandoObject`. They are related to using nullable data, editing, filtering, grouping, and nested model properties. A [runnable example](#example) is available afterwards.
+
+### Editing
+
+* If any non-string property in the `ExpandoObject` is `nullable`, then do not set [`EditorType`]({%slug components/grid/editing/overview%}#customize-the-editor-fields) for the column. Instead, use a [Grid column `EditorTemplate`]({%slug grid-templates-editor%}). The following code snippet is part of the full example below.
 
 <div class="skip-repl"></div>
 
 ````CSHTML
-<GridColumn Field="@item.Key" FieldType="@(typeof(DateTime?))" @key="@item.Key">
+<GridColumn Field="PropertyDate"
+            FieldType="@GetPropertyType("PropertyDate")"
+            Title="Date declared">
     <EditorTemplate>
         @{
             var editItem = (IDictionary<string, object>)context;
-            DateTime? dateValue = (DateTime?)(editItem[item.Key]);
+            DateTime? dateValue = (DateTime?)(editItem["PropertyDate"]);
             <TelerikDatePicker Value="@dateValue"
-                               ValueChanged="@( (DateTime? newValue) => editItem[item.Key] = newValue )"
-                               DebounceDelay="0" />
+                                ValueChanged="@( (DateTime? newValue) => editItem["PropertyDate"] = newValue )"
+                                DebounceDelay="0" />
         }
     </EditorTemplate>
 </GridColumn>
 ````
 
-* [Binding to `ExpandoObject` is not (yet) supported together with `LoadGroupsOnDemand="true"`](https://feedback.telerik.com/blazor/1535071). A possible workaround is:
-    1. Bind the Grid with its [`OnRead` event]({%slug common-features-data-binding-onread%}).
-    1. Populate the missing `MemberType` properties of the `FilterDescriptor`s in the [`OnRead` event argument]({%slug common-features-data-binding-onread%}#event-argument), namely in `args.Request.Filters`.
+### Filtering
 
->caption Using ExpandoObject and LoadGroupsOnDemand
+* Filtering in nullable data requires you to use [`OnRead` data binding]({%slug common-features-data-binding-onread%}). Populate the missing `MemberType` properties in the `CompositeFilterDescriptor`s in the [`OnRead` event argument]({%slug common-features-data-binding-onread%}#event-argument), namely in `args.Request.Filters`. Also see the [related public item](https://feedback.telerik.com/blazor/1499221-incorrect-filter-operators-when-the-fieldtype-is-set-to-a-nullable-numeric-data-type). The following code snippet is part of the full example below.
 
-````CSHTML
-@using System.Dynamic
-@using Telerik.DataSource
-@using Telerik.DataSource.Extensions
+<div class="skip-repl"></div>
```

Comment:
Indent the entire code snippet so that it gets rendered as nested under the bullet item above.